### ISSUE TITLE
Adding commands from ISCP v1.34

### ISSUE THREAD

**Issue by algorythmic** [miracle2k/onkyo-eiscp#35]
I have the latest ISCP definition and took a stab at generating the YAML representation from it but it's not so simple..  First of all, these are now distributed as .xlsx and tablib was choking on the file for some reason.

More importantly though, it seems very difficult to recreate the (very necessary) manual modifications that were made (especially in a way that would make for a clear merge).

If you want to look at the new file, you might have a better sense of how best to proceed..

**Comment by algorythmic** [comment]
I think the most useful commands to add would be those related to new inputs, since these are probably the things people are most interested in controlling:
```yaml
      '11':
        name: strmbox
        description: sets STRM BOX
      '12':
        name: tv
        description: sets TV
```
```yaml
      2D:
        name: airplay
        description: sets Aiplay
      2E:
        name: bluetooth
        description: sets Bluetooth
```
```yaml
      '33':
        name: dab
        description: sets DAB
      '55':
        name: hdmi5
        description: sets HDMI 5
      '56':
        name: hdmi6
        description: sets HDMI 6
      '57':
        name: hdmi7
        description: sets HDMI 7
```

There's also a new command to get some info back in XML format, which includes some information not otherwise available:
```yaml
  NRI:
    name: receiver-information
    description: Receiver Information
    values:
      QSTN:
        name: query
        description: gets the Receiver Information Status
```

If you're curious, this looks like:

```xml
<response status="ok">
  <device id="TX-RZ610">
    <brand>ONKYO</brand>
    <category>AV Receiver</category>
    <year>2016</year>
    <model>TX-RZ610</model>
    <destination>Dx</destination>
    <macaddress>[redacted]</macaddress>
    <modeliconurl>http://[redacted]/icon/OAVR_120.jpg</modeliconurl>
    <friendlyname/>
    <firmwareversion>1020-5010-2102-0000-0000</firmwareversion>
    <netservicelist count="6">
      <netservice id="0e" value="1" name="TuneIn Radio" account="Username" password="Password"/>
      <netservice id="04" value="1" name="Pandora" account="Email" password="Password"/>
      ....
    </netservicelist>
    <zonelist count="4">
      <zone id="1" value="1" name="Main" volmax="82" volstep="1"/>
      ....
    </zonelist>
    <selectorlist count="14">
      <selector id="10" value="1" name="Roku 4" zone="01" iconid="10"/>
      ....
    </selectorlist>
    <presetlist count="40">
      <preset id="01" band="0" freq="0" name=""/>
      ....
    </presetlist>
    <controllist count="52">
      <control id="Bass" value="1" zone="3" min="-10" max="10" step="1"/>
      ....
    </controllist>
    <functionlist count="8">
      <function id="UsbUpdate" value="0"/>
      ....
    </functionlist>
    <tuners count="2">
      <tuner band="FM" min="87500" max="107900" step="200"/>
      <tuner band="AM" min="530" max="1710" step="10"/>
    </tuners>
  </device>
</response>
````

The most useful stuff above is probably the custom names for inputs.

**Comment by miracle2k** [comment]
Very interesting, thanks. The XML could possibly be used to make the CLI UI smarter, e.g. by knowing how many zones exists, or the names of the inputs.

As for the merge, I would try to do it like this:

1. Brach off the repo from the point of the original, unmodified YAML import.
2. Replace the YAML file with the new version.
3. Rebase all the YAML changes from the  master branch on top.

**Comment by algorythmic** [comment]
Without getting `import_protocol_doc.py` to work with the xlsx file, it will be very difficult to get the new `modelsets` and populate the `models:` attribute of the new commands.

There are also some complications with the current approach to designating supported models, with some newly published commands that work only *before* a firmware update is applied and some working only after.  (Previously, there were commands that worked only after an update, but no commands that stopped working when an update was applied).  Perhaps this issue can be ignored and the module can assume that the latest FW is in use?

What I'd like to check about is if you would accept a PR that included just some subset of the new commands with a bogus `models:` attribute, or if you would only want to merge in a full set of new commands with the `models:` attribute properly populated and new `modelsets` created.  I can provide the former, but don't think I can figure out the latter.

**Comment by marcelrv** [comment]
@algorythmic  Hi I'm trying to find this latest version of ISCP definitions.
Can you share it or point where I can download it. (Hoping to find my model TX-NR656 ISCP details)

**Comment by algorythmic** [comment]
Onkyo makes these available if you request them by emailing onkyoadmin@us.onkyo.com with a subject line referencing ISCP/RS232 codes.

**Comment by toshibochan** [comment]
any update to add Airplay and strmbox input?

**Comment by abeckerman** [comment]
Here's the latest codes spec, as obtained from Onkyo this AM...

[RS-232 and IP INTEGRA (August 2016 NEW).xlsx](https://github.com/miracle2k/onkyo-eiscp/files/879596/RS-232.and.IP.INTEGRA.August.2016.NEW.xlsx)

**Comment by miracle2k** [comment]
1.34 has been merged. For any future updates, let's open a new ticket.
